src/java/org/apache/cassandra/cql3/selection/WritetimeOrTTLSelector.java

```diff
@@ -45,29 +43,30 @@ protected Selector deserialize(DataInputPlus in, int version, TableMetadata meta
             ByteBuffer columnName = ByteBufferUtil.readWithVIntLength(in);
             ColumnMetadata column = metadata.getColumn(columnName);
             int idx = in.readInt();
-            boolean isWritetime = in.readBoolean();
-            return new WritetimeOrTTLSelector(column, idx, isWritetime);
+            int ordinal = in.readByte();
```

Comment:
Good question! I think this patch is going to be merged with trunk instead of 4.1. 

Luckily, both `readBoolean` and `readByte` consume one byte. When the cluster runs in mix mode, it should be able to ser/deser w/o throwing exceptions. 
The prior logic is to use `ttl` when reads 0 and use `writetime` when reads 1. I need to adjust the `Kind` enum ordering to reflect that, i.e. `TTL: 0, WRITE_TIME: 1, MAX_WRITE_TIME: 2`. Thanks for pointing it out.
When the cluster is upgrading, we should not expect client to start sending queries that use the new function. So the value of the byte field should be either 0 or 1, which can be interpreted correctly on both old and new versions. 

Does it sound good to you?

Reply:
Sounds good to me. Indeed it's fortunate that `readBoolean` also reads a byte. Probably we should add a note in the upgrade notes in `NEWS.txt` warning about `MAX_WRITE_TIME` not working on rolling upgrade scenarios. By the way, we should also add an entry about the new feature in the section about new features of that file.